homeassistant/components/binary_sensor/template.py

```diff
@@ -148,6 +177,26 @@ def _async_render(self):
                 return
             _LOGGER.error("Could not render template %s: %s", self._name, ex)
 
+        for property_name, template in (
+                ('_icon', self._icon_template),
+                ('_entity_picture', self._entity_picture_template)):
+            if template is None:
+                continue
+
+            try:
+                setattr(self, property_name, template.async_render())
+            except TemplateError as ex:
+                friendly_property_name = property_name[1:].replace('_', ' ')
+                if ex.args and ex.args[0].startswith(
```

Comment:
I think you still want an `else` that just logs an error ([like here](https://github.com/home-assistant/home-assistant/pull/11530/files#diff-e7d0db0df3cad9b49841ae34d3dd18ddR197)) so the user is informed of invalid templates.

Reply:
I think I fixe this in the patch below